source/dbo/Stored Procedures/Power BI Reporting/uspUpdateRecordNhsNumber.sql

```diff
@@ -0,0 +1,16 @@
+﻿CREATE PROCEDURE [dbo].[uspUpdateRecordNhsNumber] AS
+	SET NOCOUNT ON
+
+	BEGIN TRY
+
+		UPDATE [dbo].[Record_PersonalDetails]
+			SET NhsNumber = SUBSTRING(NhsNumberToLookup, 1, 3) +
+				' ' + SUBSTRING(NhsNumberToLookup, 4, 3) +
+				' ' + SUBSTRING(NhsNumberToLookup, 7, 4)
+			WHERE NhsNumberToLookup NOT LIKE '%[^0-9]%'
```

Comment:
@LeoCie I think you've got two NOTs in this patindex.  The ^ means 'match any character not between the square brackets' so we're saying any NHS number which isn't not like 0 .. 9 !

We also want to make sure the string is 10 characters long before we begin, no point formatting if it isn't a valid NHS number to start with.

Reply:
Yes the double not is intentional. If I do `LIKE '%[0-9]%'` it would let through things such as `abc9102030abc`. 
I will change it so we check the length.

Reply:
I have now added in length verification

Reply:
It's frustrating that there isn't a more intuitive way to do this... but I don't see one